src/libcore/char.rs

```diff
@@ -902,6 +902,11 @@ impl<I: Iterator<Item = u8>> Iterator for DecodeUtf8<I> {
             }
         })
     }
+
+    #[inline]
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.0.size_hint()
```

Comment:
This look incorrect. This iterator can yield fewer `char`s than it has input `u8`s. See `std::str::Chars::size_hint`.

Reply:
Ah, right you are. I saw `self.0.next().map(...)` in the `next()` implementation and assumed that meant a 1:1 correspondence.